### PR TITLE
Enable CoreCLR init failure logging in single exe host

### PR DESCRIPTION
**Pull request by janvorli**
When adding the CoreCLR initialization failure logging recently, I have missed the fact that there are two versions of the coreclr_resolver.cpp. One for standalone host that I have added support for, but also for linking into single exe.

This change adds the missing support to the single exe host.

### CODE REVIEW

File: src/native/corehost/hostpolicy/static/coreclr_resolver.cpp [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include <pal.h>
 #include <trace.h>
 
+typedef void (*coreclr_error_writer_callback_fn) (const char *message);
```
Comment (vitek-karas) [inline src/native/corehost/hostpolicy/static/coreclr_resolver.cpp:11]: I don't think this is necessary - the same typedef already exists in `coreclr_resolver.h` which is included.

**Review by vitek-karas — APPROVED** [review]
Looks good - I would only remove the unnecessary typedef.

File: src/native/corehost/hostpolicy/static/coreclr_resolver.cpp (hunk 1 above)
Comment (janvorli) [reply src/native/corehost/hostpolicy/static/coreclr_resolver.cpp]: Removed in the last commit.

**Comment by janvorli** [comment]
The CI failure is a known issue unrelated to this change.

**Pull request merged**